File: frontend/my-app/src/types/product.ts
export type Product = {
  id: number;
  name: string;
  description?: string | null;
  price: number;
  stock: number;
  sku: string;
  image_url?: string | null;
  status: string;
  category_id: number;
  created_at?: string;
  updated_at?: string;
};

export type ProductInput = Omit<Product, "id" | "created_at" | "updated_at">;

File: backend/models/product.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Text,
    Numeric,
    DateTime,
    ForeignKey
)
from sqlalchemy.sql import func

from db.database import Base


class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)

    name = Column(String(255), nullable=False)

    description = Column(Text, nullable=True)

    price = Column(
        Numeric(10, 2),
        nullable=False
    )

    stock = Column(
        Integer,
        nullable=False,
        default=0
    )

    sku = Column(
        String(100),
        nullable=False,
        unique=True
    )

    image_url = Column(Text, nullable=True)

    status = Column(
        String(20),
        nullable=False,
        default="Active"
    )

    category_id = Column(
        Integer,
        ForeignKey("categories.id"),
        nullable=False
    )

    created_at = Column(
        DateTime,
        server_default=func.now()
    )

    updated_at = Column(
        DateTime,
        server_default=func.now(),
        onupdate=func.now()
    )